### PR TITLE
Add allocatable property to spaces. SPC-39 #resolve

### PR DESCRIPTION
**Pull request by pasadinhas**
Still needs integration (e.g. change minor).

SpaceClassification are not allocatable by default. This means that the SpaceClassification that can be allocated should be marked as such when this code goes to production.

### CODE REVIEW

File: src/main/resources/resources/FenixEduSpacesResources_en.properties [hunk 1]
```diff
@@ -37,3 +37,4 @@ label.spaceClassification.noSuchClass = No such class
 label.spaceClassification.typeMismatch = The declared type and does not match the selected default value
 label.start.date.is.after.end.date = The end date must be greater or equal to the start date
 label.usermismatch = User mismatch
+e
```
Comment (sergiofbsilva) [inline src/main/resources/resources/FenixEduSpacesResources_en.properties]: this change is unnecessary.

File: src/main/webapp/WEB-INF/resources/SpacesResources_en.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
 OccupationRequestState.NEW = New
 OccupationRequestState.OPEN = Open
 OccupationRequestState.RESOLVED = Resolved
-calendar.add.event = Seleccionar Per�odo
+calendar.add.event = Seleccionar Per�odo
```
Comment (sergiofbsilva) [inline src/main/webapp/WEB-INF/resources/SpacesResources_en.properties]: some encoding problem here ?

File: src/main/webapp/WEB-INF/resources/SpacesResources_pt.properties [hunk 3]
```diff
@@ -147,6 +147,7 @@ label.spaces.blueprint.number = Número da Planta
 label.spaces.blueprint.show = mostrar
 label.spaces.blueprintNumber = Nº de Planta
 label.spaces.classification = Classificação
+label.spaces.classification.allocatable = Reservavel
```
Comment (sergiofbsilva) [inline src/main/webapp/WEB-INF/resources/SpacesResources_pt.properties]: maybe Reservável ?

**Comment by pasadinhas** [comment]
Those issues should now be fixed

File: src/main/java/org/fenixedu/spaces/domain/SpaceClassification.java [hunk 4]
```diff
@@ -71,10 +71,15 @@ public int compare(SpaceClassification o1, SpaceClassification o2) {
     };
 
     public SpaceClassification(String code, LocalizedString name, SpaceClassification parent, JsonElement metadataSpec) {
+        this(code, name, parent, metadataSpec, false);
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/spaces/domain/SpaceClassification.java]: default should be true to make this releasable in a minor version.

File: src/main/java/org/fenixedu/spaces/ui/SpaceClassificationBean.java [hunk 5]
```diff
@@ -57,6 +59,7 @@ public SpaceClassificationBean() {
         this.metadataSpec = new JsonArray();
         this.parent = SpaceClassification.getRootClassification().getExternalId();
         this.code = "";
+        this.isAllocatable = false;
```
Comment (sergiofbsilva) [inline src/main/java/org/fenixedu/spaces/ui/SpaceClassificationBean.java]: maybe this should also be true ?

File: src/main/webapp/WEB-INF/resources/SpacesResources_en.properties [hunk 6]
```diff
@@ -1,7 +1,7 @@
 OccupationRequestState.NEW = New
 OccupationRequestState.OPEN = Open
 OccupationRequestState.RESOLVED = Resolved
-calendar.add.event = Seleccionar Per�odo
+calendar.add.event = Seleccionar Período
```
Comment (sergiofbsilva) [inline src/main/webapp/WEB-INF/resources/SpacesResources_en.properties]: why is this a change ? LF vs CRLF ?

File: src/main/webapp/static/fenix-spaces/js/editMetadata.js [hunk 7]
```diff
@@ -89,11 +91,12 @@ app.controller('MetadataController', [ '$scope','$timeout', '$modal', '$log','$t
 		}
 		var urlToPost = location.href;
 		debugger;
-		$http({method : 'POST', url : urlToPost, data : 
+		$http({method : 'POST', url : urlToPost, data :
```
Comment (sergiofbsilva) [inline src/main/webapp/static/fenix-spaces/js/editMetadata.js]: again CRLF vs LF ?

**Pull request merged**